File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/alarm/entity/AlarmMessage.java
package cc.nobrain.dev.userserver.domain.alarm.entity;


import cc.nobrain.dev.userserver.domain.alarm.enums.AlarmEventType;
import cc.nobrain.dev.userserver.domain.base.entity.BaseCU;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import java.util.List;


@DynamicUpdate
@Entity
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AlarmMessage extends BaseCU {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String title;

    @Column
    private String content;

    @Column
    @Enumerated(EnumType.STRING)
    private AlarmEventType eventType;

    @OneToMany(mappedBy = "alarmMessage", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<AlarmTarget> alarmTarget;

    @OneToMany(mappedBy = "alarmMessage", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<AlarmRead> alarmRead;
}

File: UserServer/src/main/kotlin/cc/nobrain/dev/userserver/domain/alarm/entity/AlarmMessage.kt
package cc.nobrain.dev.userserver.domain.alarm.entity

import cc.nobrain.dev.userserver.domain.alarm.enums.AlarmEventType
import cc.nobrain.dev.userserver.domain.base.entity.BaseCU
import jakarta.persistence.*
import org.hibernate.annotations.DynamicUpdate

@DynamicUpdate
@Entity
class AlarmMessage(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column
    var title: String? = null,

    @Column
    var content: String? = null,

    @Column(nullable = true)
    var link: String? = null,

    @Column
    @Enumerated(EnumType.STRING)
    var eventType: AlarmEventType? = null,

    @OneToMany(mappedBy = "alarmMessage", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    var alarmTarget: MutableList<AlarmTarget> = mutableListOf(),

    @OneToMany(mappedBy = "alarmMessage", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
    var alarmRead: MutableList<AlarmRead> = mutableListOf()
) : BaseCU() {
}
